### PR TITLE
[Slo Correction] Add docs for recurring slo correction

### PR DESCRIPTION
**Pull request by iminoso**
Add documentation for recurring slo correction fields.

### CODE REVIEW

**Review by athap — DISMISSED** [review]
just a nit comment 
Looks good otherwise!

File: datadog/resource_datadog_slo_correction.go [hunk 1]
```diff
@@ -35,8 +35,8 @@ func resourceDatadogSloCorrection() *schema.Resource {
 			},
 			"end": {
 				Type:        schema.TypeInt,
-				Required:    true,
-				Description: "Ending time of the correction in epoch seconds.",
+				Optional:    true,
+				Description: "Ending time of the correction in epoch seconds. Required for one time corrections, but optional if `rrule` is specified",
```
Comment (skarimo) [inline datadog/resource_datadog_slo_correction.go:39]: Can we mark this field to conflict with `rrule` and `duration`?

File: datadog/tests/resource_datadog_slo_correction_test.go [hunk 2]
```diff
@@ -132,6 +165,48 @@ func testAccCheckDatadogSloCorrectionConfig(uniq string) string {
     `, uniq, uniq)
 }
 
+func testAccCheckDatadogSloCorrectionConfigRecurring(uniq string) string {
+	return fmt.Sprintf(`
+		resource "datadog_service_level_objective" "foo" {
```
Comment (skarimo) [inline datadog/tests/resource_datadog_slo_correction_test.go:170]:
```suggestion
	resource "datadog_service_level_objective" "foo" {
```

File: datadog/resource_datadog_slo_correction.go [hunk 3]
```diff
@@ -53,6 +53,16 @@ func resourceDatadogSloCorrection() *schema.Resource {
...
 			},
+			"duration": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Description: "Length of time in seconds for a specified `rrule` recurring SLO correction (required if specifying `rrule`)",
+			},
+			"rrule": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Recurrence rules as defined in the iCalendar RFC 5545.",
```
[the remark this answers is not in the document]
Comment (iminoso) [reply datadog/resource_datadog_slo_correction.go:65]: ah we don't explicitly throw errors if things like `dtstart` and `dtend` are actually included in the `rrule` - I think we can add a follow up for that then include in this doc

File: docs/resources/slo_correction.md [hunk 4]
```diff
@@ -39,6 +39,15 @@ resource "datadog_slo_correction" "example_slo_correction" {
   slo_id      = "datadog_service_level_objective.example_slo.id"
   timezone    = "UTC"
 }
+resource "datadog_slo_correction" "example_slo_correction_with_recurrence" {
+  category    = "Scheduled Maintenance"
+  description = "correction example with recurrence"
+  start       = 1735707000
```
[the remark this answers is not in the document]
Comment (athap) [reply docs/resources/slo_correction.md:45]: ah I see, no worries then

**Review by athap — APPROVED** [review]

**Review by skarimo — APPROVED** [review]

**Pull request merged**